sklearn/utils/parallel.py

```diff
@@ -125,3 +130,35 @@ def __call__(self, *args, **kwargs):
             config = {}
         with config_context(**config):
             return self.function(*args, **kwargs)
+
+
+def _get_threadpool_controller():
+    """Return the global threadpool controller instance.
+
+    It can be used to locally limit the number of threads without looping through all
+    shared libraries every time.
+    """
+    if not hasattr(sklearn, "_threadpool_controller"):
+        sklearn._threadpool_controller = ThreadpoolController()
```

Comment:
I find it a tiny bit weird to use `sklearn._threadpool_controller`, i.e. that calling `sklearn.utils.parallel._get_threadpool_controller` has a side-effect on the `sklearn` root module. I would slightly prefer a module-level variable in `sklearn/utils/parallel.py`, which you could access with `sklearn.utils.parallel._threadpool_controller`.

I guess one of the reason is to have `sklearn._threadpool_controller` as before this PR, are there other reasons?

Reply:
You're right, I just moved it without much considerations but it should be set on the `utils.parallel` module.